src/JsonApi/Serializer/ErrorNormalizer.php

```diff
@@ -0,0 +1,60 @@
+<?php
+
+/*
+ * This file is part of the API Platform project.
+ *
+ * (c) Kévin Dunglas <dunglas@gmail.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace ApiPlatform\Core\JsonApi\Serializer;
+
+use ApiPlatform\Core\Problem\Serializer\ErrorNormalizerTrait;
+use Symfony\Component\Debug\Exception\FlattenException;
+use Symfony\Component\Serializer\Normalizer\NormalizerInterface;
+
+final class ErrorNormalizer implements NormalizerInterface
+{
+    const FORMAT = 'jsonapi';
+
+    use ErrorNormalizerTrait;
+
+    private $debug;
+
+    public function __construct(bool $debug = false)
+    {
+        $this->debug = $debug;
+    }
+
+    public function normalize($object, $format = null, array $context = [])
+    {
+        if ($this->debug) {
+            $trace = $object->getTrace();
+        }
+
+        $message = $object->getErrorMessage($object, $context, $this->debug);
+
+        $data = [
+            'title' => $context['title'] ?? 'An error occurred',
+            'description' => $message ?? (string) $object,
```

Comment:
`(string) $object` is not correct here because if `$object` is a `FlattenException` instance then you will get a fatal error because this class cannot be converted to string.

BTW, `ErrorNormalizerTrait::getErrorMessage()` always returns a string, so the null coalescing operator is useless here:
```php
'description' => $this->getErrorMessage($object, $context, $this->debug),
```